### PR TITLE
Keep accurate term in history on the primary

### PR DESCRIPTION
**Pull request by achamayou**
Next part of #2087

This is necessary so the frontend can atomically capture a (term, version, root) and initialise the Tx with an up to date (term, read_version) and the corresponding merkle tree root.

### CODE REVIEW

File: src/consensus/aft/raft.h [hunk 1]
```diff
@@ -1857,7 +1857,8 @@ namespace aft
     void become_leader()
     {
       election_index = last_committable_index();
-      LOG_DEBUG_FMT("Election index is {}", election_index);
+      LOG_DEBUG_FMT(
+        "Election index is {} in term", election_index, state->current_view);
```
Comment (ashamis) [inline src/consensus/aft/raft.h:1861]: I think you want a second {} somewhere

File: src/kv/store.h [hunk 2]
```diff
@@ -549,14 +542,24 @@ namespace kv
             compacted));
         }
 
+        // The term should always be updated on rollback() when passed
+        // regardless of whether version needs to be updated or not
+        if (t.has_value())
```
Comment (ashamis) [inline src/kv/store.h:547]: nit: i think our coding style states braces with if statements

**Review by ashamis — APPROVED** [review]

File: src/consensus/aft/raft.h (hunk 1 above)
Comment (achamayou) [reply src/consensus/aft/raft.h:1861]: Yes, thank you.

File: src/kv/store.h (hunk 2 above)
Comment (achamayou) [reply src/kv/store.h:547]: Yes, absolutely.

File: src/kv/store.h [hunk 3]
```diff
@@ -599,6 +610,9 @@ namespace kv
     {
       std::lock_guard<SpinLock> vguard(version_lock);
       term = t;
+      auto h = get_history();
+      if (h)
+        h->set_term(term);
```
Comment (eddyashton) [inline src/kv/store.h:615]:
```suggestion
      if (h)
      {
        h->set_term(term);
      }
```

**Review by eddyashton — APPROVED** [review]

**Pull request merged**